### PR TITLE
Enable recently added analyzers (and fix some violations)

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @jkoritzinsky, @buyaa-n, @Cosifne

### CODE REVIEW

File: eng/CodeAnalysis.src.globalconfig [hunk 1]
```diff
@@ -393,6 +399,9 @@ dotnet_diagnostic.CA1849.severity = suggestion
 # CA1850: Prefer static 'HashData' method over 'ComputeHash'
 dotnet_diagnostic.CA1850.severity = warning
 
+# CA1851: Possible multiple enumerations of 'IEnumerable' collection
+dotnet_diagnostic.CA1851.severity = suggestion
```
Comment (stephentoub) [inline eng/CodeAnalysis.src.globalconfig:403]: @Cosifne, @buyaa-n, I temporarily enabled this locally and fixed the things that made sense to fix, but I've left it as suggestion as there remain a non-trivial number of false positives, and it didn't seem worth it right now to suppress them each with pragmas.

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Review by Cosifne — APPROVED** [review]

**Comment by stephentoub** [comment]
Failure is https://github.com/dotnet/runtime/issues/60962

File: src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml [hunk 2]
```diff
@@ -38,11 +38,6 @@
     <type fullname="System.Threading.Timer">
       <property name="AllTimers" />
     </type>
-    <type fullname="System.Threading.ThreadPool">
-      <method name="GetQueuedWorkItemsForDebugger" />
-      <method name="GetGloballyQueuedWorkItemsForDebugger" />
-      <method name="GetLocallyQueuedWorkItemsForDebugger" />
-    </type>
```
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml:45]: Thanks, @AaronRobinsonMSFT.  That's the same takeaway I had from discussions in December.  Thanks for confirming.

[the remark this answers is not in the document]
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml:45]: Thanks for confirming.

**Pull request merged**